src/main/java/hudson/plugins/jira/JiraBuildAction.java

```diff
@@ -39,23 +42,33 @@ public String getUrlName() {
         return "jira";
     }
 
+    @Exported(inline = true)
+    public Set<JiraIssue> getIssues() {
+        return issues;
+    }
+
+    @Exported
+    public String getServerURL() {
+        JiraSite jiraSite = JiraSite.get(owner.getParent());
+        URL url = Objects.firstNonNull(jiraSite.alternativeUrl, jiraSite.alternativeUrl);
+        return url != null ? url.toString() : null;
+    }
+
     /**
      * Finds {@link JiraIssue} whose ID matches the given one.
+     * @param issueID e.g. JENKINS-1234
+     * @return JIRAIssue representing the issueID
      */
-    public JiraIssue getIssue(String id) {
+    public JiraIssue getIssue(String issueID) {
         for (JiraIssue issue : issues) {
-            if (issue.id.equals(id)) {
+            if (issue.getKey().equals(issueID)) {
                 return issue;
             }
         }
         return null;
     }
 
     public void addIssues(Set<JiraIssue> issuesToBeSaved) {
-        SortedSet<JiraIssue> allIssues = new TreeSet<JiraIssue>();
-        allIssues.addAll(issuesToBeSaved);
-        allIssues.addAll(Arrays.asList(this.issues));
-
-        this.issues = allIssues.toArray(new JiraIssue[allIssues.size()]);
+        this.issues.addAll(issuesToBeSaved);
```

Comment:
I wasn't aware that it was significant apart from using `List`. Updating a PR where we use `LinkedHashSet` instead.